core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -212,6 +214,21 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
 					log.warn(createAbortingMessage(remoteUser,405) + "method ["+method+"] not allowed");
 					return;
 				}
+
+				String authorizationHeader = request.getHeader("Authorization");
+				if(StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.contains("Bearer")) { // assumes that the jwt token is provided via authorization header
+					try {
+						Map<String, Object> claimsSet = listener.getJwtValidator().validateJWT(authorizationHeader.substring(7));
+						messageContext.setSecurityHandler(new JwtSecurityHandler(claimsSet, listener.getRoleClaim()));
+						messageContext.put("ClaimsSet", JSONObjectUtils.toJSONString(claimsSet));
+					} catch(Exception e) {
+						log.warn("unable to validate jwt",e);
+						response.sendError(401, e.getMessage());
```

Comment:
Here you should make a distinction between a JWT being cryptograpically, timewise and issuerwise invalid (401), and a cryptographically, timewise and issuerwise valid JWT that does not have the required claims (403)